elastic/kibana
# [Logs onboarding] Install system integration when onboarding system logs

Closes https://github.com/elastic/kibana/issues/163244.

### Changes
- Use `@kbn/use-tracked-promise` to fetch data from fleet apis, couldn't use the `useFetcher` hook because it's tailoring internal routes in the plugin;
- `useInstallSystemIntegration ` is a new hook responsible for installing the system integration.
- `system-integration-banner` component have been created to hold all the logic related to system integration.

#### After this changes

##### When integration is not installed

https://github.com/elastic/kibana/assets/1313018/4162319f-35d3-42d3-bd4d-821d1da26a8b

##### When integration is already installed

https://github.com/elastic/kibana/assets/1313018/5f1bf76e-7ed4-4f2c-ba4c-a8b2f3ff80a2

If a user doesn't have the required privileges to install the integrations they can still continue with the onboarding process but will see the following message in the UI. After the onboarding is finished they will be redirected to discover using dataset names `system.auth` and `system.syslog`.

<img width="2055" alt="image" src="https://github.com/elastic/kibana/assets/1313018/8214b761-1712-4c7b-888a-394e68ded59f">

### How to test?

- Enter the [custom deployment](https://yngrdyn-deploy-kiban-pr163794.kb.us-west2.gcp.elastic-cloud.com/)
- Check [installed integrations](https://yngrdyn-deploy-kiban-pr163794.kb.us-west2.gcp.elastic-cloud.com/app/integrations/installed) `/app/integrations/installed`
- Go to observability onboarding [landing page](https://yngrdyn-deploy-kiban-pr163794.kb.us-west2.gcp.elastic-cloud.com/app/observabilityOnboarding): `app/observabilityOnboarding`
- Select `Stream host system logs` or `Quickstart`
- System integration should install (if it's not installed in the deployment) or just notify that has been installed (if it's already installed in the deployment)
- After entering `Stream host system logs` page System integration should be installed
